ToggleHDR.swift

```diff
@@ -58,105 +50,121 @@ func bool(_ arg: String) -> Bool? {
+        display.setMode(originalMode)
+        return false
+    }
+    return true
 }
 
-func check60Hertz(display: MPDisplay) {
-	guard let ud = UserDefaults(suiteName: userDefaultsSuiteName) else {
-		return
-	}
-	// macOS doesn't think this display supports HDR; see if it supports HDR
-	// when set to 60 Hertz
-	guard let scanRates = display.scanRates else {
-		return
-	}
-	if scanRates.contains(where: { $0 as? Int == 60 }) {
-		guard let newMode = display.mode(matchingResolutionOfMode: display.currentMode, withScanRate: 60) as? MPDisplayMode else {
-			return
-		}
-		// We found a 60 Hertz version of the current display mode. Before
-		// we make any changes, store the CURRENT display mode in User
-		// Defaults so that we can restore it later.
-		let modeNumber = display.currentMode.modeNumber
-		ud.setValue(modeNumber, forKey: "SDRModeNumber-\(display.name)")
-        // Store the current mode so that if the 60 Hertz mode doesn't support
-        // HDR we can switch back.
-		let originalMode = display.currentMode
-		// Now switch to the 60 Hertz version of the current display mode.
-		display.setMode(newMode)
-		if !display.hasHDRModes {
-			// Even at 60 Hertz, HDR isn't supported; reset back to the
-			// original mode.
-			display.setMode(originalMode)
-		}
-	}
+func printHelp() {
+    print("""
+    Usage: \(CommandLine.arguments[0]) [id/uuid/name/all] [on/off] [--try-60-hz]
+
+    Options:
+    • id/uuid/name: The display to toggle HDR on/off for.
+    • all: Toggle HDR on/off for all displays that support it.
+    • on/off: Enable or disable HDR. If not provided, the opposite of the current state is used.
+    • --try-60-hz: If the display doesn't support HDR at the current refresh rate, it will try switching to 60 Hertz.
+
+    """)
 }
 
 func main() {
     guard let mgr = MPDisplayMgr(), let displays = mgr.displays else { return }
 
-	guard CommandLine.arguments.count >= 2 else {
-		if displays.count == 1 {
-			// If there is only one display, toggle the HDR on that display.
-			toggleHDR(display: displays[0])
-			return
-		}
+    guard CommandLine.arguments.count >= 2 else {
+        if displays.count == 1, displays[0].hasHDRModes {
+            // If there is only one display, toggle the HDR on that display.
+            toggleHDR(display: displays[0])
+            return
+        }
+        printHelp()
+        printDisplays(displays)
+        return
+    }
+
+    if CommandLine.arguments.contains("--help") || CommandLine.arguments.contains("-h") {
+        printHelp()
         printDisplays(displays)
-        print("\nUsage: \(CommandLine.arguments[0]) [id/uuid/name/all] [on/off]")
         return
     }
 
     defer {
         print("")
         printDisplays(displays)
     }
-    let arg = CommandLine.arguments[1].lowercased()
+    let try60Hz = CommandLine.arguments.contains("--try-60-hz")
+    let args = CommandLine.arguments.filter { arg in !["--try-60-hz"].contains(arg) }
+    let arg = args[1].lowercased()
```

Comment:
Won't this crash if `--try-60-hz` is the only command line argument?

Reply:
Sorry for using my work account for these two comments. I can confirm here that they both come from the original PR's submitter (me).

Reply:
Indeed, I'll fix it. Thanks for catching it!